### DISCUSSION TITLE
Can I use mermaid.js?

### DISCUSSION THREAD

**Discussion by mcauto** [swagger-api/swagger-ui#8303] [Q&A] [answered]
I want to use mermaid diagram in description of openapi.yaml

https://mermaid-js.github.io/mermaid/#/./n00b-gettingStarted

**Comment by char0n** [comment, marked as answer]
@mcauto unfortunately you cannot. SwaggerUI uses https://www.npmjs.com/package/remarkable which only supports [CommonMark](http://commonmark.org/) as mandated by OpenAPI specification.

SwaggerUI is plugginable system thought, thus it should be possible to create plugin that overrides how MarkDown is rendered.

**Reply by mathis-m** [reply]
@mcauto you would just need to override the `Markdown` component as shown [here under the components segment](https://swagger.io/docs/open-source-tools/swagger-ui/customization/plugin-api/).  
Just use the `source` argument that gets passed to the component and render it using `react-mermaid2`:

```ts
import React from "react"
import Mermaid from "react-mermaid2"

const CustomMarkdown: React.FC<{ source: string }> = ({source}) => {
  return (
    <Mermaid chart={source} />
  );
};

export default React.memo(CustomMarkdown);
```
